zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/IdentitySetupInitializeProcessor.java

```diff
@@ -41,108 +36,63 @@ public final class IdentitySetupInitializeProcessor
   private final KeyGenerator keyGenerator;
   private final TypedCommandWriter commandWriter;
   private final StateWriter stateWriter;
-  private final RoleState roleState;
 
-  public IdentitySetupInitializeProcessor(
-      final Writers writers,
-      final KeyGenerator keyGenerator,
-      final ProcessingState processingState) {
+  public IdentitySetupInitializeProcessor(final Writers writers, final KeyGenerator keyGenerator) {
     this.keyGenerator = keyGenerator;
     commandWriter = writers.command();
     stateWriter = writers.state();
-    roleState = processingState.getRoleState();
   }
 
   @Override
   public void processRecord(final TypedRecord<IdentitySetupRecord> command) {
     final var initializationKey = keyGenerator.nextKey();
     final var setupRecord = command.getValue();
 
-    final var defaultRole = setupRecord.getDefaultRole();
-    final var roleExists = roleState.getRole(defaultRole.getRoleId()).isPresent();
-    createDefaultRole(initializationKey, defaultRole);
-
-    final var defaultTenant = setupRecord.getDefaultTenant();
-    createDefaultTenant(initializationKey, defaultTenant);
-
-    final var users = setupRecord.getUsers();
-    createUsers(initializationKey, users, defaultRole, defaultTenant);
-
-    final var mappings = setupRecord.getMappings();
-    createMappings(initializationKey, mappings, defaultRole, defaultTenant);
-
-    if (!roleExists) {
-      addAllPermissions(initializationKey, defaultRole);
-    }
+    createRoles(initializationKey, setupRecord.getRoles());
+    createDefaultTenant(initializationKey, setupRecord.getDefaultTenant());
+    createUsers(initializationKey, setupRecord.getUsers());
+    createMappings(initializationKey, setupRecord.getMappings());
+    createRoleMembers(initializationKey, setupRecord.getRoleMembers());
+    createTenantMembers(initializationKey, setupRecord.getTenantMembers());
+    createAuthorizations(initializationKey, setupRecord.getAuthorizations());
 
     stateWriter.appendFollowUpEvent(
         initializationKey, IdentitySetupIntent.INITIALIZED, setupRecord);
   }
 
-  private void createDefaultRole(final long key, final RoleRecord defaultRole) {
-    commandWriter.appendFollowUpCommand(key, RoleIntent.CREATE, defaultRole);
+  private void createDefaultTenant(final long key, final TenantRecord defaultTenant) {
```

Comment:
🙃  this is peak pedantry so not going to push at all but I like the structure in the method declarations (considering for access levels too) to be the order in which they are used, an example is that `createRoles` is further down this list but called first.